services/src/main/java/org/keycloak/protocol/oidc/OIDCWellKnownProvider.java

```diff
@@ -166,6 +168,7 @@ public Object getConfig() {
         config.setResponseModesSupported(DEFAULT_RESPONSE_MODES_SUPPORTED);
         config.setGrantTypesSupported(DEFAULT_GRANT_TYPES_SUPPORTED);
         config.setAcrValuesSupported(getAcrValuesSupported(realm));
+        config.setPromptValuesSupported(DEFAULT_PROMPT_VALUES_SUPPORTED);
```

Comment:
IMHO this should be realm specific. Due to that, the list of default prompt values should only contain those that are allowed for this realm. 

That is: Only if registration is allowed in this realm, the prompt `create` should be included in the list. 

Not sure if there could be conditions for the other values.

Reply:
I made the generation of `prompt_values_supported` realm aware and only add the `create` scope when registrations are allowed for the realm.

Reply:
Thank you, this resolves it for me. I'll leave it for the core-client team to do a more in-depth review.